### PR TITLE
COTCO-536: improved security around the missing provider id

### PR DESCRIPTION
**Pull request by robertbutacu**

### CODE REVIEW

File: app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala [hunk 1]
```diff
@@ -26,10 +26,10 @@ trait CTCError extends Product with Serializable {
 }
 
 case class MalformedRequest(message: String = "Malformed request") extends CTCError
-case class MongoDBError(message: String) extends CTCError
-case class MongoDBNoResults(message: String) extends CTCError
+case class MongoDBError(message: String)        extends CTCError
```
Comment (udinobi) [inline app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: Robert... just my 2 cents and u do not need to change anything here, but please do not go crazy for justifying all our code base  ;)

File: app/uk/gov/hmrc/ctc/services/TransitService.scala [hunk 2]
```diff
@@ -48,27 +44,37 @@ class TransitService @Inject()(transitRepository: TransitRepository, appConfig:
...
 
-  def nextTransit(destinationPort: DestinationPort, transitState: TransitSlot)(implicit hc: HeaderCarrier): ServiceResponse[TransitData] = {
+  def nextTransit(destinationPort: DestinationPort, transitState: TransitSlot)(implicit headerCarrier: HeaderCarrier): ServiceResponse[TransitData] = {
     val inactivity = appConfig.transitInactivityTimeout
-    val pid = providerId(hc)
+    val pid = providerId(headerCarrier)
 
-    val result = transitState.value match {
-      case "active" => transitRepository.getClosestActiveTransit(destinationPort, inactivity, pid)
+    def result(pid: String): ServiceResponse[Transit] = transitState.value match {
```
Comment (udinobi) [inline app/uk/gov/hmrc/ctc/services/TransitService.scala]: please, for the sake of uniformity, rename `pid` to `providerId`

File: app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala (hunk 1 above)
Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: Im not sure I fully understand - is it about CTCError or something else?

Comment (udinobi) [reply app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: It was not specifically related to this line. In general I see that you always try to justify/indent (by using blanks) a straight set of lines when they contain common logic. I am aware this is a controversial topic, but still I think we can read the code even without indentation.

Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: Ah, sure - its a matter of styling, I prefer pattern matching/if/case classes to be aligned, but point taken

**Pull request closed without merging**